cms/plugin_rendering.py

```diff
@@ -71,7 +71,11 @@ def render_plugin(context, instance, placeholder, template, processors=None, cur
         content = render_to_string(template, flatten_context(context))
     elif (isinstance(template, Template) or (hasattr(template, 'template') and
           hasattr(template, 'render') and isinstance(template.template, Template))):
-        content = template.render(context)
+        if type(context) == PluginContext:
```

Comment:
no need for the typecheck as `PluginContext` is a `Context` subclass.

`context = context.flatten()`